### ISSUE TITLE
Non exact matching broken when using expects()

### ISSUE THREAD

**Issue by AlexStansfield** [mockery/mockery#827]
The `expects()` method seems to have an issue when you want to use it with non exact call counts.

The code (https://github.com/mockery/mockery/blob/master/library/Mockery/Mock.php#L263) calls `once()` on the Expectation which adds an exact count validator for `1` to the `_countValidators`. 

Then if you want to use `between`, `atLeast`, `zeroOrMoreTimes` those validators are added to the `_countValidators` after the exact validator. It checks the exact count match against `1` first and fails immediately. 

I've had to switch to `allows()` rather than `expects()` to get around this.

**Comment by robertbasic** [comment]
Yes, that is the expected behaviour of `expects()` http://docs.mockery.io/en/latest/reference/alternative_should_receive_syntax.html#expects as we are expecting an exact number of calls to that method.

On the other hand, `allows()` allows a varying number of calls to a method.

**Comment by AlexStansfield** [comment]
The docs don't really make that clear at all.

`We use expects() when we want to verify that a particular method was called`

and

`By default expects() sets up an expectation that the method should be called once and once only. If we expect more than one call to the method, we can change that expectation.`

Reading that would make me think that I could use `atLeast`, `Between`, etc. And it did work that way until this change - https://github.com/mockery/mockery/commit/683cf51db3e8bf10cbebb005b7ae3b81f1ba79b4

**Comment by robertbasic** [comment]
We thought that the distinction is clear enough from the actual method names: `expects` vs `allows`.

The things is that `atLeast` and `between` allow a method to be called from zero to n times. `expects` expects a call, one by default, or if we change that with `twice` or `times`, that many times.

If nothing else, I'll update the docs to make this distinction between allows and expects even clearer.

Thanks for bringing this issue to our attention!

**Comment by chriys** [comment]
@robertbasic any help needed for the doc ?

**Comment by robertbasic** [comment]
@chriys always :)

**Comment by davedevelopment** [comment]
I think we could allow this again, we should be able to allow them to do 

```
$mock->expects()->foo('bar')->atLeast()->twice();
```

I'll take a look later.

**Comment by adam-ja** [comment]
I realise this is an old issue but since it's still open, is this ever likely to change? @davedevelopment suggested that it should be allowed and I agree - saying a method call is "expected at least n times" reads much better than saying it is "allowed at least n times" in my opinion.
